### PR TITLE
fix: compress data from API for the page load performance improvement

### PR DESCRIPTION
**Pull request by keita-determined**
## Description
[WEB-1870]

The response data from (almost?) all determined AI API endpoints were not compressed at all, so that caused long response time with `Fast 3G network` (look at `Content download` time). 

Especially,  `experiments-search` API usually returns big data, so this PR improve the page loading time around **10x**.

### Before

Screenshot 2024-01-19 at 12 27 22 PM

---

Screenshot 2024-01-19 at 1 08 27 PM

### After

Screenshot 2024-01-19 at 12 25 09 PM

---

Screenshot 2024-01-19 at 1 07 29 PM

## Test Plan
- Check if API data are compressed
- Check if there's `Content-Encoding: gzip` in response header
- Check if `js/css/html` files are still compressed

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[WEB-1870]: https://hpe-aiatscale.atlassian.net/browse/WEB-1870?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/core.go [hunk 1]
```diff
@@ -966,10 +966,15 @@ func (m *Master) Run(ctx context.Context, gRPCLogInitDone chan struct{}) error {
 	// Other static files should only be cached for a short period of time
 	cacheFileShortTerm := regexp.MustCompile(`.(antd.\S+(.css)|ico|png|jpe*g|gif|svg)$`)
 
+	// API endpoints
+	apiRegex := regexp.MustCompile(`^/api/.+$`)
+
 	gzipConfig := middleware.GzipConfig{
 		Skipper: func(c echo.Context) bool {
```
Comment (keita-determined) [inline master/internal/core.go:973]: question: is there something we don't wanna compress? I followed the current code, but not sure if we want `Skipper` here

Comment (eecsliu) [reply master/internal/core.go:973]: hmmm I'm not familiar with compression... @maxrussell would you have any thoughts?

Comment (maxrussell) [reply master/internal/core.go:973]: Nothing I can think of. Seems like a good change.

**Review by maxrussell — APPROVED** [review]

File: master/internal/core.go (hunk 1 above)
Comment (keita-determined) [reply master/internal/core.go:973]: ok, i'll remove `Skipper` and relavant variables, then merge this PR. thanks for the review

Comment (keita-determined) [reply master/internal/core.go:973]: After some investigation, i conclude that `Skipper` is still required for gzip.

i found that `test_gc_checkpoints_lfs` test case should avoid being compressed on CI (look at [CI error](https://app.circleci.com/pipelines/github/determined-ai/determined/50208/workflows/5e7dd7b5-a455-4277-a575-beab314c49ee/jobs/2207487/parallel-runs/3))

Also, according to @azhou-determined, 
> gzip didn’t play nicely with the tensorboard profiler plugin. Tensorboard wouldn’t load when we compressed the responses

**Pull request merged**